### PR TITLE
Player events refactor

### PR DESCRIPTION
**Pull request by Dananji**
Related issue: #5473

### CODE REVIEW

File: app/assets/javascripts/player_listeners.js [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+ * Unless required by applicable law or agreed to in writing, software distributed
+ *   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ---  END LICENSE_HEADER BLOCK  ---
+*/
+
+let canvasIndex = -1;
+let currentSectionLabel = undefined;
+let listenersAdded = false;
```
Comment (cjcolvar) [inline app/assets/javascripts/player_listeners.js:19]: Is this used only for add to playlist listeners?  If so how about renaming this variable?
```suggestion
let addToPlaylistListenersAdded = false;
```

**Review by cjcolvar — APPROVED** [review]
As far as I can tell this looks good.  Great job cleaning this up and consolidating it!

File: app/assets/javascripts/player_listeners.js (hunk 1 above)
Comment (Dananji) [reply app/assets/javascripts/player_listeners.js:19]: Yes, that sounds good 👍

**Comment by Dananji** [comment]
@cjcolvar I re-named the variable and brought in the changes we did in the old code to fix enabling/disabling add to playlist button in iPad.

**Review by cjcolvar — APPROVED** [review]
:+1:

**Pull request merged**